clang/lib/CIR/CodeGen/CIRGenExprAgg.cpp

```diff
@@ -136,7 +136,39 @@ class AggExprEmitter : public StmtVisitor<AggExprEmitter> {
   void VisitPointerToDataMemberBinaryOperator(const BinaryOperator *E) {
     llvm_unreachable("NYI");
   }
-  void VisitBinAssign(const BinaryOperator *E) { llvm_unreachable("NYI"); }
+  void VisitBinAssign(const BinaryOperator *E) {
```

Comment:
Nice! Please add missing asserts/unreacheable before the first call to `buildLValue`:
- hasSameUnqualifiedType assertion.
- `isBlockVarRef(E->getLHS()) && E->getRHS()->HasSideEffects(CGF.getContext())`.

Quick reminder: we should always assert/unreachable for things we don't implement, this should make it easier for people adding new features to easily spot where they should be added.